index.js

```diff
@@ -23,6 +23,37 @@ function router (controller, routes, options) {
     state.set(urlStorePath, input.route.url);
   }
 
+  function getUrl(route, input) {
+    var url = route;
+    var params = route.match(/:.[^\/]*/g);
+
+    if (params) {
+      // If called from a url change, add params and query to input
+      if (input.route && input.route.params) {
+        input = params.reduce(function (input, param) {
+          var key = param.substr(1, param.length);
+          input[key] = input.route.params[key];
+          return input;
+        }, input);
+      }
+
+      // Create url based on direct signal input or
+      // params passed from addressbar
+      url = params.reduce(function (url, param) {
+        var key = param.substr(1, param.length);
+        if (!(key in input)) {
+          throw new Error('Cerebral router - The signal "' + routes[route] + '" is bound to "' + route + '" route, but required param "' + key + '" wasn\'t provided.');
```

Comment:
Can't find a good way to do it. If I catch an error in exposed `getUrl` I am loosing information about which key wasn't provided. I think a message is generic enough to be thrown in both cases.